dna/tests/src/project/project_init.rs

```diff
@@ -0,0 +1,84 @@
+#[cfg(test)]
+pub mod tests {
+    use crate::project::fixtures::fixtures::ProjectMetaFixturator;
+    use ::fixt::prelude::*;
+    use hdk::prelude::*;
+    use hdk_crud::WrappedAgentPubKey;
+    use holochain_types::prelude::option_entry_hashed;
+    use holochain_types::prelude::ElementFixturator;
+    use holochain_types::prelude::ValidateDataFixturator;
+    use projects::project::error::Error;
+    use projects::project::project_meta::crud::ProjectMeta;
+    use projects::project::project_meta::validate::*;
+    use projects::init;
+
+    #[test]
+    fn test_init() {
+               
+        let mut mock_hdk = MockHdkT::new();
+        
+        let zome_info = fixt!(ZomeInfo);
+        // do these mock functions need to be called in the same order as the actual code?
+        // init calls create_receive_signal_cap_grant, which calls zome_info and create_cap_grant
+        mock_hdk
+            .expect_zome_info()
+            .times(1)
+            .return_const(Ok(zome_info.clone())); // define what it should return when it matches the `with` statement, or matches at all (without `.with`)
+
+        // create_cap_grant calls just `create` under the hood
+        let mut functions: GrantedFunctions = BTreeSet::new();
+        functions.insert((zome_info.zome_name, "recv_remote_signal".into()));
+        // expected is for the .with, and is b/c create parameter is of type EntryWithDefId
+        let expected = EntryWithDefId::new(
+            EntryDefId::CapGrant,
+            Entry::CapGrant(CapGrantEntry {
+                tag: "".into(),
+                access: ().into(),
+                functions,
+            }),
+        );
+        let header_hash = fixt!(HeaderHash);
+        mock_hdk
+            .expect_create()
+            .with(mockall::predicate::eq(expected))
+            .times(1)
+            .return_const(Ok(header_hash.clone()));
```

Comment:
great! you even got the `.with` which requires mocking a suitable match for the input type. It does strengthen the nature of the test

Reply:
I actually found a test function for `create_receive_signal_cap_grant` in your crud_hdk crate, and just copied that.

Reply:
that's totally cool. Resourcefulness is key

Reply:
:wink: